openzwave-sys/build.rs

```diff
@@ -1,4 +1,64 @@
 extern crate gcc;
+use std::fs;
+use std::path::Path;
+use std::process::Command;
+use std::env;
+
+fn find_include_path(c: &mut gcc::Config) -> Result<(), ()> {
+    // TODO try to actually build with these paths
+    let possible_paths = vec![
+        "/usr/include/openzwave",
+        "/use/local/include/openzwave",
+    ];
+
+    let some_header = "Manager.h";
+
+    let include_path = try!(
+        possible_paths
+            .iter()
+            .find(|device_name| fs::metadata(Path::new(device_name).join(some_header)).is_ok())
+            .map(|&str| str)
+            .ok_or(())
+    );
+
+    c.include(include_path);
+    Ok(())
+}
+
+#[cfg(target_os = "linux")]
+fn target_specific_work() {
+    println!("cargo:rustc-link-lib=udev");
+    println!("cargo:rustc-link-lib=static=openzwave");
+}
+
+#[cfg(target_os = "macos")]
+fn target_specific_work() {
+    // The .a is a universal (fat) binary, so let's convert it to a thin binary
+    // There is no easy way to disable the fat binary generation in open-zwave:
+    // https://github.com/OpenZWave/open-zwave/issues/814
+    let exit_code = Command::new("lipo")
+        .current_dir("open-zwave")
+        .arg("-thin").arg("x86_64")
+        .arg("-output").arg("libopenzwave-thin.a")
+        .arg("libopenzwave.a")
+        .status()
```

Comment:
Why does it matter? Won't the linker just pick the bits that it wants? I'm not a Mac developer, so this is more a question than a comment.